server/apps/immich/src/api-v1/share/share.core.ts

```diff
@@ -0,0 +1,76 @@
+import { SharedLinkEntity } from '@app/database/entities/shared-link.entity';
+import { CreateSharedLinkDto } from './dto/create-shared-link.dto';
+import { ISharedLinkRepository } from './shared-link.repository';
+import crypto from 'node:crypto';
+import { BadRequestException, InternalServerErrorException, Logger } from '@nestjs/common';
+import { AssetEntity } from '@app/database';
+
+export class ShareCore {
+  readonly logger = new Logger(ShareCore.name);
+
+  constructor(private sharedLinkRepository: ISharedLinkRepository) {}
+
+  async createSharedLink(userId: string, dto: CreateSharedLinkDto): Promise<SharedLinkEntity> {
+    try {
+      const sharedLink = new SharedLinkEntity();
+
+      sharedLink.id = Buffer.from(crypto.randomBytes(70));
+      sharedLink.key = Buffer.from(crypto.randomBytes(50));
```

Comment:
Why do we need two sets of crypto bytes? Can we not consolidate these into a single column and use just one value?